boardswarm-cli/src/ui.rs

```diff
@@ -70,79 +74,40 @@ impl Terminal {
     }
 }
 
-async fn process_input<R>(mut input: R) -> (Bytes, R)
-where
-    R: AsyncRead + Unpin,
-{
-    let mut buffer = [0; 4096];
-    let read = input.read(&mut buffer).await.unwrap();
-    (Bytes::copy_from_slice(&buffer[0..read]), input)
-}
-
 #[derive(Debug, Clone)]
 enum Input {
-    PowerOn,
-    PowerOff,
-    PowerReset,
     Up,
     Down,
     ScrollReset,
-    Bytes(Bytes),
     UIMessage(String),
 }
 
-struct InputStream<R> {
-    saw_escape: bool,
-    future: tokio_util::sync::ReusableBoxFuture<'static, (Bytes, R)>,
-}
-
-impl<R> InputStream<R>
-where
-    R: AsyncRead + Unpin + Send + 'static,
-{
-    fn new(rx: R) -> Self {
-        let future = ReusableBoxFuture::new(process_input(rx));
-        Self {
-            saw_escape: false,
-            future,
+fn convert_keycode_to_term_code(code: KeyCode, modifiers: KeyModifiers) -> Option<Vec<u8>> {
+    let val = match code {
+        KeyCode::Esc => vec![0x1B_u8],
+        KeyCode::Enter => vec![b'\n'],
+        KeyCode::Tab => vec![b'\t'],
+        KeyCode::Backspace => vec![0x8_u8],
+        KeyCode::Insert => b"\x1B[2~".to_vec(),
+        KeyCode::Delete => b"\x1B[3~".to_vec(),
+        KeyCode::PageUp => b"\x1B[5~".to_vec(),
+        KeyCode::PageDown => b"\x1B[6~".to_vec(),
+        KeyCode::Home => b"\x1B[1~".to_vec(),
+        KeyCode::End => b"\x1B[4~".to_vec(),
+        KeyCode::Up => b"\x1B[A".to_vec(),    // CUU
+        KeyCode::Down => b"\x1B[B".to_vec(),  // CUD
+        KeyCode::Left => b"\x1B[D".to_vec(),  // CUB
+        KeyCode::Right => b"\x1B[C".to_vec(), // CUF
+        KeyCode::Char('c') if modifiers == KeyModifiers::CONTROL => vec![0x3_u8], // ^C ETX
+        KeyCode::Char('d') if modifiers == KeyModifiers::CONTROL => vec![0x4_u8], // ^D EOT
+        KeyCode::Char(c) => {
```

Comment:
Rather then writing he ansi codes by hand; why not use crossterms Command where possible ? Or hopefully there are other crates to do the translation.

Also are you sure this is complete set?